src/package.ts

```diff
@@ -0,0 +1,57 @@
+import { privateEncrypt } from 'crypto';
+import contextKeys from './contextKeys';
+import { exec, pathExists } from './utilities';
+
+export class SPMPackage {
+	private constructor(
+        readonly folder: string,
+        public contents: any
```

Comment:
Can we try to avoid using `any`? 

We could, for example, define interfaces declaring the fields we expect to find in the JSON. These don't have to be complete, but they would remove a lot of mystery from the code.

Reply:
Fair point

Reply:
Added basic interfaces, they can be expanded in a later PR if required. As I added an interface for `Target` I replaced the one in `SwiftTaskProvider.ts`